src/osmf/lib/HDSPlayer.as

```diff
@@ -750,7 +750,7 @@ package
         totalTime = 0;
       }
       eventObject.currentTime = _mediaPlayerSprite.mediaPlayer.currentTime;
-      eventObject.duration = duration
+      eventObject.duration = totalTime;
```

Comment:
Sat with @mityaha to see it working with the other change. Looks to be working fine. Merging the changes.